erigon-lib/common/mem/mem_linux.go

```diff
@@ -92,7 +104,10 @@ func LogVirtualMemStats(ctx context.Context, logger log.Logger) {
 				continue
 			}
 
-			v := VirtualMemStat{memStats}
+			var m runtime.MemStats
+			dbg.ReadMemStats(&m)
```

Comment:
Note that `ReadMemStats` is not considered "virtual memory" so there is a bit of mixing of concepts going on in this function called "LogVirutalMemStats". Also I believe `ReadMemStats` is re-usable across os-es while currently `LogVirtualMemStats` is only for Linux atm. I would suggest to add another func "LogMemStats" which calls `ReadMemStats` + `ReadMemVirtualMemStats` and logs a combined view of the 2. This is more extendable in the future if we are to add more metrics.

Reply:
Maybe that is ok? - nope